api/v1beta1/rollout_types.go

```diff
@@ -75,6 +75,100 @@ type RolloutStrategy struct {
 	Paused bool `json:"paused,omitempty"`
 	// +optional
 	Canary *CanaryStrategy `json:"canary,omitempty"`
+	// +optional
+	BlueGreen *BlueGreenStrategy `json:"blueGreen,omitempty" protobuf:"bytes,1,opt,name=blueGreen"`
+}
+
+// Get the rolling style based on the strategy
+func (r *RolloutStrategy) GetRollingStyle() RollingStyleType {
+	if r.BlueGreen == nil && r.Canary == nil {
+		return EmptyRollingStyle
```

Comment:
we should include a webhook to ensure that either bluegreen or canary field is set

Reply:
done